### DISCUSSION TITLE
Labels placed from cropped video

### DISCUSSION THREAD

**Discussion by HallaMo** [DeNardoLab/BehaviorDEPOT#17] [Q&A]
Hi,

I analyze my videos in deeplabcut by setting "crop" to true, this leads to more accurate position data. BehaviorDEPOT does not have any information about my cropping parameters, so when I'm running the software, I notice that the labels are placed according to the cropped parameters. Is there a way to correct that issue?

**Comment by cjgabrie** [comment]
Hello! Thanks for trying out BehaviorDEPOT.

I haven't encountered this issue before, but it should be
relatively straightforward to fix. Would you be able to send an example of
your DLC output file? It would be useful to see a sample CSV and H5 (they
can be from the same files) that have the cropping problem.

Thanks,
Chris

**Comment by HallaMo** [comment]
Sure, but unfortunately it seems like I can only upload the csv file here, I hope that's enough for you? It's also quite visible in the created trajectory map which I also attached. 

I should probably also add that the video I use for analysis had the mpg video format, but since BehaviorDEPOT requires mp4 or avi, I converted the video. I will use the analyse function on the converted video and see whether that has any effect.

[Animal408_OF_2023_09_07DLC_resnet50_FCroomSep12shuffle1_250000.csv](https://github.com/DeNardoLab/BehaviorDEPOT/files/12701477/Animal408_OF_2023_09_07DLC_resnet50_FCroomSep12shuffle1_250000.csv)
![Spatial Trajectory Map](https://github.com/DeNardoLab/BehaviorDEPOT/assets/82214406/c50937c3-fa59-4f59-a02c-76a9f5bf943c)

**Comment by HallaMo** [comment]
Hi, is there any update on this? Using the analyse function on the converted video unfortunately did not help.

**Comment by cjgabrie** [comment]
Hello, apologies for the delay--

Unfortunately, the CSV file did not provide any immediate solutions. To the best of my understanding of the issue, DLC should be recording coordinates in uncropped space instead of the relative coordinates you currently have. If provided with the cropping parameters, I could replicate the cropping within BehDEPOT, but that information is not recorded in the output file.

Are you able to make labeled videos using DLC successfully?

It seems that you could be dealing with this issue: https://github.com/DeepLabCut/DeepLabCut/issues/1979

If so, one solution could be to use dynamic cropping instead of manual cropping parameters. This setting doesn't seem to be affected by the coordinate issue.

Otherwise, if you can provide a file that has the cropping parameters, I could try to work in a fix that applies the same cropping within BD.

**Comment by HallaMo** [comment]
Hi,

I usually use cropping since this increases the accuracy of the position outcome data and was thinking that better coordinates also lead to better behDepot analysis results. I don't have the issue in deeplabcut itself when creating videos (see attachment), probably because the config file contains the cropping parameters.
The parameters I use are the following:
x1: 350
x2: 950
y1: 20
y2: 600

I haven't used dynamic cropping so far but will test that today, maybe that's a good way to solve the issue.

![Screenshot (63)](https://github.com/DeNardoLab/BehaviorDEPOT/assets/82214406/571fa58a-2d54-494f-958f-5a11928b3e95)
